### PR TITLE
feat: better print `retif`

### PR DESCRIPTION
**Pull request by evilebottnawi**

### CODE REVIEW

File: tests/member_chain/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -171,17 +171,20 @@ $object[$valid
...
+    : $helper->responseBody($this->defaultUser)
+)
     ->map()
     ->filter();
 
 ($valid
     ? $helper->responseBody($this->currentUser)
-    : $helper->responseBody(defaultUser))->map();
+    : $helper->responseBody(defaultUser)
+)->map();
```
Comment (evilebottnawi) [inline tests/member_chain/__snapshots__/jsfmt.spec.js.snap]: A `prettier` do  :star:

File: tests/retif/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -93,4 +120,59 @@ return (
...
+$var =
+    'string' .
+    $someOtherReallyReallyLongVariable .
+    $someOtherReallyReallyLongVariable
+        ? 1
+        : 2;
+$var = 1
+    ? 'string' .
+        $someOtherReallyReallyLongVariable .
+        $someOtherReallyReallyLongVariable
```
Comment (evilebottnawi) [inline tests/retif/__snapshots__/jsfmt.spec.js.snap]: Indent :star:

File: src/printer.js [hunk 3]
```diff
@@ -2158,14 +2158,65 @@ function printStatement(path, options, print) {
       ]);
     }
     case "retif": {
-      const printedExpr = concat([
+      const parts = [];
+      const parent = path.getParentNode();
+      const parentParentNode = path.getParentNode(1);
```
Comment (czosel) [inline src/printer.js]: If this is the parent's parent, should we call this `grandParentNode`?

File: src/printer.js [hunk 4]
```diff
@@ -2158,14 +2158,65 @@ function printStatement(path, options, print) {
...
+      const part = concat([
         line,
         "?",
-        node.trueExpr ? concat([" ", path.call(print, "trueExpr"), line]) : "",
+        node.trueExpr
+          ? concat([
+              " ",
+              node.trueExpr.kind === "bin"
+                ? indent(printedTrueExpr)
+                : path.call(print, "trueExpr"),
```
Comment (czosel) [inline src/printer.js]: Replace with `printedTrueExpr`?

File: src/printer.js [hunk 5]
```diff
@@ -2158,14 +2158,65 @@ function printStatement(path, options, print) {
...
       ]);
-      return group(concat([path.call(print, "test"), indent(printedExpr)]));
+
+      parts.push(part);
+
+      // We want a whole chain of retif to all break if any of them break.
+      const maybeGroup = doc =>
+        parent === firstNonRetifParent ? group(doc) : doc;
+
+      // Break the closing paren to keep the chain right after it:
```
Comment (czosel) [inline src/printer.js]: `paren` -> `parens`?

File: src/printer.js [hunk 6]
```diff
@@ -2158,14 +2158,65 @@ function printStatement(path, options, print) {
...
+      // ($a
+      //   ? $b
+      //   : $c
+      // )->call()
+      const breakLookupNodes = ["propertylookup", "staticlookup"];
+      const breakClosingParen =
```
Comment (czosel) [inline src/printer.js]: `breakClosingParens`?

File: src/printer.js (hunk 3 above)
Comment (evilebottnawi) [reply src/printer.js]: @czosel looks as good name, but in prettier and other places we use `parentParentNode`, i think better left as is

File: src/printer.js (hunk 4 above)
Comment (evilebottnawi) [reply src/printer.js]: :+1:

File: src/printer.js (hunk 5 above)
Comment (evilebottnawi) [reply src/printer.js]: :+1:

File: src/printer.js (hunk 6 above)
Comment (evilebottnawi) [reply src/printer.js]: :+1:

**Comment by evilebottnawi** [comment]
@czosel ready for review

**Pull request merged**